indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2285,7 +2385,9 @@ private void checkTaskDuration() throws ExecutionException, InterruptedException
         }
         // clear partitionGroups, so that latest sequences from db is used as start sequences not the stale ones
         // if tasks did some successful incremental handoffs
-        partitionGroups.get(groupId).replaceAll((partition, sequence) -> getNotSetMarker());
+        for (PartitionIdType partitionId : group.startingSequences.keySet()) {
```

Comment:
nit: I see this loop repeated a few times, maybe good to make a method

Reply:
hm, it's small enough that I'll leave it be for now